csrc/scheduler/transpose.cpp

```diff
@@ -117,11 +156,10 @@ class DomainMap : public pointwise_utils::DomainMap {
     const auto& dom = tv->getLeafDomain();
     for (auto i : c10::irange(dom.size())) {
       if (dom[i] == mapped_id) {
-        return i;
+        return static_cast<int64_t>(i);
       }
     }
-    TORCH_INTERNAL_ASSERT(
-        false, "Can not find ID mapped to ", root_dim, " in tensor ", tv);
+    return -1;
```

Comment:
This mapping here could and will fail with resize.

i.e. The example i run into earlier was when reference have a split on constant factor. We'll ended up with trying to map `i0` to `i1, 4, ceilDiv((i0*i2), 4)`.

More importantly, I think our scheduling expects an exact mapping of inner most dimension. So rejecting this without trying to map it.

Reply:
Sorry, I'm not sure what you meant. Should we remove the support of `resize` in line 153?

Reply:
err. sorry I meant `reshape` instead of `resize`. I'm just trying to explain why we would want to allow `getInnerLeafDim` to fail the mapping and return a `-1` here.

Reply:
OK, but still I'm not sure what you meant by:

> We'll ended up with trying to map i0 to i1, 4, ceilDiv((i0*i2), 4).

Is `i0` a root domain? What does `i1, 4, ceilDiv((i0*i2), 4)` mean? Is it the leaf domain of this tensor? Where does `i1` come from then?

Reply:
Sorry for the vague comment. Let's see if an example would help:

With a simple math:
```
T9_l[ iS37{i0}, iS38{i2}, iS39{i3}, iS40{i4} ]
   = T5_g[ iS21{i0}, iS22{i2}, iS23{i3}, iS24{i4} ]
   * T8_g[ iS33{i0}, iS34{i2}, iS35{i3}, bS36{1} ];
T10_g[ iS47{( i0 * i2 )}rf, iS51{2}rf, iS53{4}rf, iS54{( ceilDiv(( ceilDiv(( i3 * i4 ), 2) ), 4) )}rf ] = view( T9_l[ iS37{i0}, iS38{i2}, iS39{i3}, iS40{i4} ] )
```

Because of the broadcast on T8 on fastest dim, we are seeing this tensor has a different fastest inner dimension `i3`.

```
T5_g[ iS21{i0}, iS22{i2}, iS23{i3}, iS24{i4} ]
 root domain : (iS21{i0}, iS22{i2}, iS23{i3}, iS24{i4})
 contiguity: t t t t
 leaf domain : (iS21{i0}, iS22{i2}, iS23{i3}, iS24{i4})
T8_g[ iS33{i0}, iS34{i2}, iS35{i3}, bS36{1} ]
 root domain : (iS33{i0}, iS34{i2}, iS35{i3}, bS36{1})
 contiguity: t t t n
 leaf domain : (iS33{i0}, iS34{i2}, iS35{i3}, bS36{1})
T9_l[ iS37{i0}, iS38{i2}, iS39{i3}, iS40{i4} ]
 root domain : (iS37{i0}, iS38{i2}, iS39{i3}, iS40{i4})
 contiguity: t t t t
 leaf domain : (iS37{i0}, iS38{i2}, iS39{i3}, iS40{i4})
T10_g[ iS47{( i0 * i2 )}rf, iS51{2}rf, iS53{4}rf, iS54{( ceilDiv(( ceilDiv(( i3 * i4 ), 2) ), 4) )}rf ]
 root domain : (iS45{i0}rf, iS46{i2}rf, iS48{i3}rf, iS49{i4}rf)
  Merge: iS45{i0}rf and iS46{i2}rf -> iS47{( i0 * i2 )}rf
  Merge: iS48{i3}rf and iS49{i4}rf -> iS50{( i3 * i4 )}rf
  Outer split: iS50{( i3 * i4 )}rf by factor 2 -> iS51{2}rf, iS52{( ceilDiv(( i3 * i4 ), 2) )}rf, start offset: 0, stop offset: 0
  Outer split: iS52{( ceilDiv(( i3 * i4 ), 2) )}rf by factor 4 -> iS53{4}rf, iS54{( ceilDiv(( ceilDiv(( i3 * i4 ), 2) ), 4) )}rf, start offset: 0, stop offset: 0
 rfactor domain : (iS47{( i0 * i2 )}rf, iS51{2}rf, iS53{4}rf, iS54{( ceilDiv(( ceilDiv(( i3 * i4 ), 2) ), 4) )}rf)
 contiguity: t t t t
 leaf domain : (iS47{( i0 * i2 )}rf, iS51{2}rf, iS53{4}rf, iS54{( ceilDiv(( ceilDiv(( i3 * i4 ), 2) ), 4) )}rf)
```

If we take T10 as global reference, we'll be trying to project the inner most from T8 to T10, where we have:
```
(gdb) print root_dim->toString(0)
$1 = "iS35{i3}"
(gdb) print getMappedRootDimIn(tv, root_dim)->toString(0)
$14 = "iS48{i3}rf"
```

We won't be able to map it to the leaf domain, since i3 is the merge `outer`. `Merge: iS48{i3}rf and iS49{i4}rf -> iS50{( i3 * i4 )}rf`

Basically, mapping from an ID in root domain to leaf domain wouldn't necessarily succeed.

Reply:
This fails the test in `NVFuserTest.FusionReshapePersistentShmoo_CUDA`

Reply:
Thanks. Now it makes sense. 

So the initial `mapped_id` is an outer domain of a `merge`, we won't map it to the `merge` output. I wonder what should happen if `mapped_id` is merged as an outer domain with a size-1 broadcast domain. It seems it's still considered to be an inner domain, doesn't it?

Reply:
Good point. I think we should identify the special case of mapping across merge/split with size-1 broadcast domain. Similar to how we changed the behavior in `permissive_relaxed_resize_nodes_` (which is to be renamed as `inner_nodes_` per your suggestion.)

Reply:
I think at this point a merge should be just used for reshape, and we don't merge size-1 domains but just squeeze them. So I think merge with size-1 domains should never show up here. If that's true, let's add an assertion.